apps/obsidian/src/index.ts

```diff
@@ -27,6 +28,32 @@ export default class DiscourseGraphPlugin extends Plugin {
     this.addRibbonIcon("telescope", "Toggle Discourse Context", () => {
       this.toggleDiscourseContextView();
     });
+
+    this.registerEvent(
+      this.app.workspace.on("editor-menu", (menu: Menu, editor: Editor) => {
+        if (!editor.getSelection()) return;
+
+        menu.addItem((menuItem) => {
+          menuItem.setTitle("Turn into Discourse Node");
+          menuItem.setIcon("file-type");
+
+          // Create submenu using the unofficial API pattern
+          // @ts-ignore - setSubmenu is not officially in the API but works
```

Comment:
Will this be supported?  What is obsidian's stance on editing the submenu?

Reply:
i couldn't find any official documentation to create submenu but followed this forum discussion:
https://forum.obsidian.md/t/submenu-items-within-the-editor-context-menu/39699